sdk/agent/ingest.go

```diff
@@ -0,0 +1,80 @@
+package agent
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/stellar/experimental-payment-channels/sdk/state"
+)
+
+var ingestingFinished = errors.New("ingesting finished")
+
+func (a *Agent) ingest() error {
+	tx, ok := <-a.streamerTransactions
+	if !ok {
+		return ingestingFinished
+	}
+
+	a.mu.Lock()
+	defer a.mu.Unlock()
```

Comment:
I suspect the biggest problem with this change ingesting every tx on network is not that additional work, but that it will frequently contest the agent's lock on the mutex. Frequent locks on this mutex will make the mutex less available for regular transactions so we may find that it affects benchmarks. I don't want to do anything about that right now, that's something we can investigate in the next milestone when we produce the benchmark. If the benchmark is unimpressive we can go profiling and looking for where contention or bottlenecks are. If it's here, then we can revisit what I tried to implement earlier in this PR's history: ingesting only transactions for the escrow accounts.